drivers/sensor/lm75/lm75.c

```diff
@@ -127,6 +339,42 @@ int lm75_init(const struct device *dev)
 	}
 #endif
 
+	ret = lm75_reg_write(dev, LM75_REG_CONFIG, &cfg->config_dt.reg, sizeof(cfg->config_dt.reg));
+	if (ret < 0) {
+		LOG_ERR("failed to write configuration (ret %d)", ret);
+		return ret;
+	}
+
+#if LM75_TRIGGER_SUPPORT
+	data->dev = dev;
+	k_work_queue_start(&data->workq, data->stack, K_THREAD_STACK_SIZEOF(data->stack),
+			   CONFIG_LM75_TRIGGER_THREAD_PRIO, NULL);
+	k_thread_name_set(&data->workq.thread, "lm75_trigger");
+	k_work_init(&data->work, lm75_trigger_work_handler);
+
+	if (cfg->int_gpio.port != NULL) {
+		if (!device_is_ready(cfg->int_gpio.port)) {
+			LOG_ERR("INT GPIO not ready");
+			return -EINVAL;
+		}
+
+		ret = gpio_pin_configure_dt(&cfg->int_gpio, GPIO_INPUT);
+		if (ret < 0) {
+			LOG_ERR("failed to configure INT GPIO (ret %d)", ret);
+			return ret;
+		}
+
+		gpio_init_callback(&data->int_gpio_cb, lm75_int_gpio_callback_handler,
+				   BIT(cfg->int_gpio.pin));
+
+		ret = gpio_add_callback(cfg->int_gpio.port, &data->int_gpio_cb);
+		if (ret < 0) {
+			LOG_ERR("failed to add INT GPIO callback (ret %d)", ret);
+			return ret;
+		}
+	}
```

Comment:
It does not seem like this driver will work with `LM75_TRIGGER_SUPPORT` enabled and without `int_gpio` pin defined, yet there's a path to gracefully initialize in that scenario.

```suggestion
		if (!device_is_ready(cfg->int_gpio.port)) {
			LOG_ERR("INT GPIO not ready");
			return -EINVAL;
		}

		ret = gpio_pin_configure_dt(&cfg->int_gpio, GPIO_INPUT);
		if (ret < 0) {
			LOG_ERR("failed to configure INT GPIO (ret %d)", ret);
			return ret;
		}

		gpio_init_callback(&data->int_gpio_cb, lm75_int_gpio_callback_handler,
				   BIT(cfg->int_gpio.pin));

		ret = gpio_add_callback(cfg->int_gpio.port, &data->int_gpio_cb);
		if (ret < 0) {
			LOG_ERR("failed to add INT GPIO callback (ret %d)", ret);
			return ret;
		}
```

Reply:
#74342 74342

Reply:
As pointed out here https://github.com/zephyrproject-rtos/zephyr/pull/74342#discussion_r1645904160, this suggestion wasn't accurate and won't go in the follow-up PR.